### DISCUSSION TITLE
Зависает

### DISCUSSION THREAD

**Discussion by KIPOWEC** [vvip-68/GyverPanelWiFi#244] [Задавайте вопросы]
Приветствую всех, столкнулся с такой ситуацией гирлянда прекрасно работает пока есть WIFI, если WIFI нет, то гирлянда постоянно подвисает на пару секунд в течении минуты пару раз использую версию WiFiPanel-v.1.08 на Wemos D1 mini. Подскажите как исправить.

**Comment by vvip-68** [comment]
Отключить все ресурсы, которые требуют доступа к интернету (получение времени из интернет, получение погоды из интернет, не использовать управление через MQTT).  Других вариантов нет.

**Reply by KIPOWEC** [reply]
Спасибо, буду пробовать.

**Reply by megadenis** [reply]
Аналогичная проблема, но я не связываю это с Wi-Fi. Прошил нормально работающую несколько лет гайвер лампу (16*16) с оригинальной прошивкой на сабжевую (последняя доступная), подвисают эффекты + как будто недостаточно кадров, анимация двигается микрорвыквами. MQTT, часы, погода, управление через DMX активированы. В приложении не настоен MQTT, т.е не используется, часы и погоду выключил. Сабжевая прошивка (правда пердыдущая версия) работает на другой панели размером 4 на 1 метр на 450 светодиодов - работает отлично, но с хорошим покрытием Wi-Fi (с автивированными часами, погодой, MQTT) - подскажите пожалуйста куда копать? БП нормальный, подвисают даже простые эффекты, особенно заметно в момент смены направления движения анимации, подвисает даже огонь, подвисают ползунки яркости. Из дополнительного, подключена тач кнопка (инвертирована А0, в прошивке активирована обычная, т.к. при активации сеносорной - у меня не завелась).

**Reply by vvip-68** [reply]
Если у вас MQTT в прошивке активирован, но не настроен - это не отменяет того, что прошивка будет куда-то лезть с дефолтными значениями MQTT, замирать на ожидании ответа (когда ничего другого не выполняется), по таймауту получать отлуп и продолжать выполнение скетча. Что внешне и проявляется как замирания и рывки.

В настройках соединения в приложении PanelSetup  - канал MQTT отключен?

**Reply by megadenis** [reply]
В WiFI Panel Setup чекбокс установлен только на Локальная сеть.

**Reply by vvip-68** [reply]
Тогда нет повода к зависаниям.
Скриншот этой странички настроек можно посмотреть?

**Reply by megadenis** [reply]
![Screenshot_20221214_070344](https://user-images.githubusercontent.com/23151954/207510475-0d7edd86-1f13-44ba-9dbf-76a88348d179.jpg)

**Reply by vvip-68** [reply]
Ну вот видите - я вам про одно приложение говорю и его настройки - про **PanelSetup** - вы мне скриншот совсем с другого приложения присылаете.

**Reply by megadenis** [reply]
Все аналогично 
![Screenshot_20221214_070852](https://user-images.githubusercontent.com/23151954/207511048-e0b8d669-29ab-4d6c-95b7-16e33f6aa085.jpg)

**Reply by megadenis** [reply]
![Screenshot_20221214_071032](https://user-images.githubusercontent.com/23151954/207511302-fa73d55a-35af-4fde-8515-54843a675d7b.jpg)

**Reply by vvip-68** [reply]
И что? Вы мне второй раз присылаете скриншот страницы подключения к матрице - то есть настройки как ваш телефон подключается к матрице. Я же в очередной раз прошу скриншот страницы настроек подключения матрицы к сети.

**Reply by vvip-68** [reply]
Точнее - приложение правильное Setup - но мне нужна не страница настроек подключения телефона к матрице, а **страница настроек подключения матрицы к сети**. Так правильнее выразиться.

**Reply by megadenis** [reply]
Я не совсем понимаю, что требуется. Страница первоначальной конфигурации в режиме точки доступа?

**Reply by vvip-68** [reply]
Требуется:
1. Открыть приложение 'WiFi Panel Setup' (то что на последнем скриншоте) и подключиться к матрице.
2. В верхнем тулбаре появятся кружочки со значками - это страницы настройки матрицы. Нажать на розовый кружок с гаечным ключем (последний в наборе в левой части тулбара).
3. Протянуть страницу, чтобы в нижней ее части был виден чекбокс "Использовать MQTT" и снять с него галочку, если она там стоит.
4. Нажать на кнопку "Сохранить", расположенную ниже этого блока

**Reply by megadenis** [reply]
Спасибо большое! Вы абсолютно правы, в данном меню как раз был включен MQTT. Видимо, он был включен там по умолчанию, в результате активации в прошивке! Буду наблюдать за работой лампы. Ещё раз спасибо, за классный проект!

**Comment by Simafor47** [comment]
Добрый день.У меня возможно похожая проблема,но наоборот,пока синхронизация не произошла работает норм,как только нажимаю кнопку в приложении "установить текущее время" зависает и ни на что не реагирует. При этом в вифи подключается и айпи получает,mqtt тоже всё ок. Куда комапь? Если при включении синхронизация включена была сразу,то горит первый диод и всё.

**Reply by KIPOWEC** [reply]
я сам дилетант подсказать не могу причину, таких проблем как у вас на моей гирлянде нет

**Comment by greeg38** [comment]
Здравствуйте, подскажите , пожалуйста, в чем дело: 
версия платы LOLIN (WEMOS) D1 D2 & mini, если поставить NodeMCU 1,0 - ничего не работает, на матрице светятся два диода, на плате не светится 
а проблема вот в чем:
не может обновить время автоматически, раз в 40 секунд зависает, все остальное работает:

12:27:56.232 -> Подключение к wifi -> 192.168.0.100
12:27:56.279 -> .............
12:28:00.304 -> WiFi подключен. IP адрес: 192.168.0.100
12:28:00.304 -> 
12:28:00.304 -> UDP-сервер на порту 2390
12:28:00.304 -> Переход в авторежим отключен
12:28:00.351 -> Режим: Дождь
12:28:10.319 -> Не удалось получить IP aдрес сервера NTP -> ru.pool.ntp.org -> 255.255.255.255
12:28:10.319 -> Используем сервер по умолчанию: 85.21.78.91
12:28:10.319 -> NTP-сервер ru.pool.ntp.org -> 85.21.78.91
12:28:10.366 -> Отправка NTP пакета на сервер ru.pool.ntp.org
12:28:15.358 -> Таймаут NTP запроса!
12:28:50.356 -> Не удалось получить IP aдрес сервера NTP -> ru.pool.ntp.org -> 255.255.255.255
12:28:50.356 -> Используем сервер по умолчанию: 85.21.78.91

и т. д.

если в прошивке указать получение погоды, вообще виснет наглухо, даже приложение не дает подключить, в мониторе порта каждые 10 сек запрос погоды

**Reply by vvip-68** [reply]
Обеспечьте выход в интернет с микроконтроллера и всё заработает. Если нет возможности выхода в интернет - отключите в прошивке (или в настройках) получение времени и погоды из интернет.

**Reply by vvip-68** [reply]
> LOLIN (WEMOS) D1 D2 & mini, если поставить NodeMCU 1,0 - ничего не работает, на матрице светятся два диода, на плате не светится
Не на тот пин припаяли вывод сигнала на матрицу. И/ИЛИ используете не ту версию ядра.

**Reply by greeg38** [reply]
> Обеспечьте выход в интернет с микроконтроллера и всё заработает. Если нет возможности выхода в интернет - отключите в прошивке (или в настройках) получение времени и погоды из интернет.

интернет есть, я раздаю его с этого же телефона
и соответственно управляю матрицей через интернет (или я чтото неправильно понимаю?)

**Reply by vvip-68** [reply]
Вероятнее всего - неправильно. Нарисуйте схему как у вас сеть сформирована, а то не очень понятно. Какие IP адреса имеют устройства в вашей сети. Почему вы уверены, что если раздаете интернет с телефона, то роутинг пакетов настроен правильно, и пакеты в интернет, отправляемые матрицей уходят дальше вашего телефона. И уж на 99%, то что вы управляете матрицей через интернет - неверно.

**Reply by greeg38** [reply]
> > LOLIN (WEMOS) D1 D2 & mini, если поставить NodeMCU 1,0 - ничего не работает, на матрице светятся два диода, на плате не светится
> > Не на тот пин припаяли вывод сигнала на матрицу. И/ИЛИ используете не ту версию ядра.

спасибо, да, действительно вывод на матрицу был D4, кнопка D3, сделал матрицу D2, кнопку D4 - все заработало с nodemcu 1.0

**Reply by greeg38** [reply]
> Вероятнее всего - неправильно. Нарисуйте схему как у вас сеть сформирована, а то не очень понятно. Какие IP адреса имеют устройства в вашей сети. Почему вы уверены, что если раздаете интернет с телефона, то роутинг пакетов настроен правильно, и пакеты в интернет, отправляемые матрицей уходят дальше вашего телефона. И уж на 99%, то что вы управляете матрицей через интернет - неверно.

Вы абсолютно правы
А схема проста, на работе вайфая нет, но есть телефон, подключенный к интернету через мобильную передачу данных, на нем включена точка доступа wifi, к которой подключается esp8266
На этом же телефоне установлены приложения wifi panel setup и  wifi panel
ip телефона из монитора порта: 192.168.43.215:2390 
ip esp8266: 192.168.43.96

вчера пробовал подключить это дело к домашнему вайфаю (который к слову тоже на постоянной основе раздается другим телефоном, да нету у нас нормального проводного интернета) )

свой телефон с установленными приложениями тоже подключил к этой сети

результат тот же - esp не может синхронизировать время

**Reply by vvip-68** [reply]
Выше в логах у вас ip esp8266 - 192.168.0.100? а это в другой сети нежели сеть вашего роутера/телефона
Что-то вы путаетесь в показаниях...
Давайте для начала актуальный лог, чего пишет есп-шка при старте

**Reply by greeg38** [reply]
> Выше в логах у вас ip esp8266 - 192.168.0.100? а это в другой сети нежели сеть вашего роутера/телефона Что-то вы путаетесь в показаниях... Давайте для начала актуальный лог, чего пишет есп-шка при старте

извиняюсь, выложил тот лог после перепрошивки, вот то что сейчас:
12:48:55.541 -> WiFiPanel v.1.13.2022.12214
12:48:55.541 -> Версия EEPROM: 0x5C
12:48:55.541 -> Host: 'WiFiPanel'
12:48:55.541 -> 
12:48:55.541 -> Инициализация файловой системы... 
12:48:55.541 -> Использовано 16384 из 2072576 байт
12:48:55.541 -> 
12:48:55.541 -> Загрузка строк выполнена.
12:48:55.541 -> Использованы адреса EEPROM 1000 - 1037
12:48:55.541 -> Свободно ячеек 3058
12:48:55.541 -> 
12:48:55.541 -> Матрица: плоская 16x16
12:48:55.541 -> Адресация: по подключению
12:48:55.541 ->   Угол: левый нижний
12:48:55.541 ->   Направление: вниз
12:48:55.541 ->   Тип: зигзаг
12:48:55.541 -> 
12:48:55.541 -> Доступные возможности:
12:48:55.541 -> + Бегущая строка: шрифт 5x8
12:48:55.541 -> + Кнопка управления: тактовая
12:48:55.541 -> + Синхронизация времени с сервером NTP
12:48:55.541 -> - Управление питанием
12:48:55.541 -> - Получение информации о погоде
12:48:55.541 -> - Управление по каналу MQTT
12:48:55.541 -> - Групповая синхронизация по протоколу E1.31
12:48:55.541 -> - Дополнительный индикатор TM1637
12:48:55.541 -> - Эффекты Jinx! с SD-карты
12:48:55.541 -> - Поддержка MP3 Player
12:48:55.541 -> 
12:48:55.541 -> Поиск файлов карт индексов...
12:48:55.541 -> Нет файлов карт индексов
12:48:55.541 -> 
12:48:55.541 -> 
12:48:55.541 -> Найденные анимации:
12:48:55.541 ->    Сердце	8x8
12:48:55.541 ->    Марио	16x16
12:48:55.541 ->    Погода	16x16
12:48:55.541 -> 
12:48:55.541 -> Поиск слайдов на FS в папке '16p16'...
12:48:55.541 -> Слайды не найдены. Будут использованы картинки эффекта 'Погода'
12:48:55.541 -> 
12:48:55.853 -> 
12:48:55.853 -> Подключение к wifi -> 192.168.43.96
12:48:55.853 -> ............................
12:49:03.356 -> WiFi подключен. IP адрес: 192.168.43.96
12:49:03.356 -> 
12:49:03.356 -> UDP-сервер на порту 2390
12:49:03.762 -> Переход в авторежим отключен
12:49:03.762 -> Настройки сохранены в EEPROM
12:49:13.427 -> Не удалось получить IP aдрес сервера NTP -> ru.pool.ntp.org -> 255.255.255.255
12:49:13.427 -> Используем сервер по умолчанию: 85.21.78.91
12:49:13.427 -> NTP-сервер ru.pool.ntp.org -> 85.21.78.91
12:49:13.427 -> Отправка NTP пакета на сервер ru.pool.ntp.org
12:49:18.440 -> Настройки сохранены в EEPROM
12:49:18.440 -> Таймаут NTP запроса!
12:49:30.776 -> UDP << ip='192.168.43.215:2390'; cmd='$18 1;'
12:49:30.776 -> Получен UDP пакeт размером 6
12:49:30.776 -> UDP 192.168.43.215:2390 >> $18 UP:0|FM:32832|AL:0|W:16|H:16|DM:0|PS:1|PD:60|IT:0|RM:1|PW:10000|BR:42|WZ:0|FS:1|EE:0;
12:49:36.002 -> UDP << ip='192.168.43.215:2390'; cmd='$18 0;'
12:49:36.002 -> Получен UDP пакeт размером 6
12:49:36.002 -> Ответ на 192.168.43.215:2390 >> ack0;
12:49:41.197 -> UDP << ip='192.168.43.215:2390'; cmd='$18 0;'
12:49:41.197 -> Получен UDP пакeт размером 6
12:49:41.197 -> Ответ на 192.168.43.215:2390 >> ack1;
12:49:53.412 -> Не удалось получить IP aдрес сервера NTP -> ru.pool.ntp.org -> 255.255.255.255
12:49:53.600 -> Используем сервер по умолчанию: 85.21.78.91
12:49:53.600 -> NTP-сервер ru.pool.ntp.org -> 85.21.78.91
12:49:53.600 -> Отправка NTP пакета на сервер ru.pool.ntp.org
12:49:53.600 -> UDP << ip='192.168.43.215:2390'; cmd='$18 0;'
12:49:53.600 -> Получен UDP пакeт размером 6
12:49:53.600 -> Ответ на 192.168.43.215:2390 >> ack2;
12:49:53.600 -> UDP << ip='192.168.43.215:2390'; cmd='$18 0;'
12:49:53.600 -> Получен UDP пакeт размером 6
12:49:53.600 -> Ответ на 192.168.43.215:2390 >> ack3;
12:49:58.467 -> Таймаут NTP запроса!
12:50:33.461 -> Не удалось получить IP aдрес сервера NTP -> ru.pool.ntp.org -> 255.255.255.255
12:50:33.461 -> Используем сервер по умолчанию: 85.21.78.91
12:50:33.461 -> NTP-сервер ru.pool.ntp.org -> 85.21.78.91
12:50:33.461 -> Отправка NTP пакета на сервер ru.pool.ntp.org
12:50:38.473 -> Таймаут NTP запроса!
12:51:13.478 -> Не удалось получить IP aдрес сервера NTP -> ru.pool.ntp.org -> 255.255.255.255
12:51:13.478 -> Используем сервер по умолчанию: 85.21.78.91
12:51:13.822 -> NTP-сервер ru.pool.ntp.org -> 85.21.78.91
12:51:13.822 -> Отправка NTP пакета на сервер ru.pool.ntp.org
12:51:18.504 -> Таймаут NTP запроса!
12:51:53.504 -> Не удалось получить IP aдрес сервера NTP -> ru.pool.ntp.org -> 255.255.255.255
12:51:53.504 -> Используем сервер по умолчанию: 85.21.78.91
12:51:53.878 -> NTP-сервер ru.pool.ntp.org -> 85.21.78.91
12:51:53.878 -> Отправка NTP пакета на сервер ru.pool.ntp.org
12:51:58.531 -> Таймаут NTP запроса!

**Reply by vvip-68** [reply]
Не вижу в логах ничего криминального.
Как проверить прохождение пакетов в интернет от esp через ваш телефон далее в интернет - я не знаю.
Разбирайтесь с таблицей роутинга в своей сети, добивайтесь, чтобы пакеты уходили в сеть далее и получали ответы из интернета, тогда заработает.
Я на этом всё. Более ничем помочь не могу.

**Comment by Raifart** [comment]
Всем привет, склепал гирлянду на wemos (esp8266) но постоянно мерцала, наткнулся на сообщение что все дело в кривости работы fastled с новым ядром 8266, что за ядро к сожалению мне никто не объяснил, программное (прошивка или что там еще в контроллере) или физическое сам контроллер.... 
**ну короче:**
взял esp 32 - вообще глушняк.

короче загрузил все... напряжение питания по сравнению с esp8266 пришлось подбирать заново, и приложение подключается только по локалке и все равно как будто с тормозами дикими, то все эффекты ичезнут из приложения то еще что, очень странно, я думал приложение и микроконтроллер общаются короткими сообщениями, а тут прям глюки самого приложения в телефоне будто бы, периодически реакции со стороны гирлянды нет, а потом как будто два эффекта разом или совершенно непонятные эффекты, но мерцания гирлянды нет :rolleyes: одно победил, другое всплыло.
через Mqtt ничего вообще не получается, хотя все настройки в прошивке сконфигурированы.
